src/neuroglancer/ui/tool.ts

```diff
@@ -211,7 +213,10 @@ export class ToolBinder extends RefCounted {
   bindings = new Map<string, Borrowed<Tool>>();
   changed = new Signal();
   private activeTool: Owned<ToolActivation>|undefined;
-  private debounceDeactivate = this.registerCancellable(debounce(() => this.deactivate(), 1));
+  private queuedTool: ToolActivation|undefined;
```

Comment:
Just make this a `Tool` rather than `ToolActivation` since the `ToolActivation` disposers will already be run when the tool is first deactivated.

Reply:
Good call, I have made this change.